src/PerfView/OtherSources/Linux/LinuxPerfScriptEventParser.cs

```diff
@@ -262,7 +263,21 @@ internal static string GetFileNameWithoutExtension(string path)
             var end = path.LastIndexOf('.');
             if (end < start)
                 end = path.Length;
-            return path.Substring(start, end - start);
+            var name = path.Substring(start, end - start);
+
+            if (stripNiSuffix)
+            {
+                var suffixStart = name.IndexOf(NISuffix);
```

Comment:
I am surprised this works as you intend because NISuffix =".ni." (that is it has a trailing dot).  

I would have expected that typically you would have a path = Foo.ni.dll and name would be Foo.ni (that is no trailing dot), and thus name.IndexOf(NISuffix) would FAIL, and the new code will never do anything.  

Are we sure that actually works (it puts both Foo AND Foo.ni in the table)?

Reply:
The input path is for the map file and not for the NI.  An example path looks like "traces/perfdemo.trace/System.Console.ni.{d2de1c76-d19d-4c52-8ff5-70e88349ea91}.map"  Originally, this function reduced this to "System.Console.ni.{d2de1c76-d19d-4c52-8ff5-70e88349ea91}".  If you specify true for stripNiSuffix then you get "System.Console.{d2de1c76-d19d-4c52-8ff5-70e88349ea91}".  For .NET Core 2.0, the perfinfo-$pid.map file no longer contains the ".ni" and that's where things were breaking.

I have confirmed that it does work, and that we do put both Foo and Foo.ni in the table.  You can see where we put both into the table here: https://github.com/Microsoft/perfview/pull/436/files#diff-151b5622877b35646b2ea5e3c9466af6R701